### ISSUE TITLE
Backpropagate through certain connections in neural networks

### ISSUE THREAD

**Issue by duguyue100** [Theano/Theano#3170]
Hi, all,

I'm implementing a WTA ConvAE that states in this paper: http://arxiv.org/pdf/1409.2752v2.pdf.
In the paper, it states that:
> we identify the single largest hidden activity within each feature map, and set the rest of the activities as well as their derivatives to zero.

and 
> the output using only the active hidden units in the feature maps and the reconstruction error is only backpropagated through these hidden units as well.

Thankfully, Theano offered `maxpool_same_size` in the new build, so I don't have to implement the first part. But for the second part, "backpropagated through certain hidden unit", I'm not so sure how is this work in Theano. Because from my understanding, for example, when I apply Dropout, I can simply mask my activation with some dropout mask, and by using the standard `tensor.grad()` method, I will not backpropagate through those connections that is zero. For this task, I thought it is the same idea. However, I'm not able to reproduce the similar results. So I'm wondering if the backpropagation has the problem.

by the way, this is the configuration of my code:
https://github.com/duguyue100/telaugesa/blob/master/scripts/cifar10_convae_test.py#L53-L87

Regards,
Hu Yuhuang.

**Comment by lamblin** [comment]
I'm sorry, but we are not able to provide that kind of support. Tickets are used for bug reports and feature requests only.
I would encourage you to post to the `theano-users` mailing list, or on StackOverflow with the `theano` tag (there is probably a machine learning tag as well).
